code/game/objects/items/food/snacks.dm

```diff
@@ -168,3 +168,79 @@
 	foodtypes = TOXIC
 	food_flags = FOOD_FINGER_FOOD
 	w_class = WEIGHT_CLASS_SMALL
+
+/obj/item/food/peanuts
+	name = "\improper Parker's unsalted peanuts"
+	desc = "As commonly seen on shuttle flights."
+	icon_state = "peanuts"
+	trash_type = /obj/item/trash/peanuts
+	food_reagents = list(/datum/reagent/consumable/nutriment = 2)
+	tastes = list("peanuts" = 3)
+	foodtypes = JUNKFOOD | NUTS
+	w_class = WEIGHT_CLASS_SMALL
+	grind_results = list(/datum/reagent/consumable/peanut_butter = 5, /datum/reagent/consumable/cooking_oil = 2)
+
+/obj/item/food/peanuts/flavoured
+	name = "\improper Parker's peanuts"
+	desc = "As commonly seen on shuttle flights."
+
+/obj/item/food/peanuts/flavoured/Initialize()
+	. = ..()
+	var/peanut_flavour = pick("salted", "wasabi", "honey-roasted", "barbecue")
+	switch(peanut_flavour)
+		if("salted")
+			name = "\improper Parker's salted peanuts"
+			desc = "As commonly seen on shuttle flights. Tastes salty, just like most peanuts."
+			food_reagents = list(/datum/reagent/consumable/nutriment = 2, /datum/reagent/consumable/salt = 1)
```

Comment:
and here, im not even sure this will work if you change food_reagents after initialize?